addons/loyalty/models/loyalty_program.py

```diff
@@ -51,12 +60,12 @@ class LoyaltyProgram(models.Model):
         ('current', 'Current order'),
         ('future', 'Future orders'),
         ('both', 'Current & Future orders')], default='current', required=True,
-         compute='_compute_from_program_type', readonly=False, store=True,
+         compute='_compute_from_program_type', readonly=True, store=True,
```

Comment:
I would rather keep it readonly=False and put it readonly in the views.
Allow advanced users to modify it.